### PR TITLE
Ignore route add failures

### PR DESCRIPTION
**Pull request by russellb**
Does the route get automatically added with IPv6 because of router
advertisements?

Also added more error details, otherwise this error is just "Exit
status 2". Similar paths need cleaning up too

### CODE REVIEW

File: go-controller/pkg/cluster/management-port_linux.go [hunk 1]
```diff
@@ -68,9 +74,14 @@ func CreateManagementPort(nodeName string, localSubnet *net.IPNet, clusterSubnet
 	}
 
 	// Create a route for the services subnet.
-	_, _, err = util.RunIP("route", "add", config.Kubernetes.ServiceCIDR, "via", routerIP)
+	_, stderr, err := util.RunIP("route", "add", config.Kubernetes.ServiceCIDR, "via", routerIP)
```
Comment (danwinship) [inline go-controller/pkg/cluster/management-port_linux.go]: This change shouldn't be needed; the service subnet isn't obviously located on this interface so there's no way the route could have been automatically added

File: go-controller/pkg/cluster/management-port_linux.go [hunk 2]
```diff
@@ -55,9 +56,14 @@ func CreateManagementPort(nodeName string, localSubnet *net.IPNet, clusterSubnet
 		}
 
 		// Create a route for the entire subnet.
-		_, _, err = util.RunIP("route", "add", subnet, "via", routerIP)
+		_, stderr, err := util.RunIP("route", "add", subnet, "via", routerIP)
 		if err != nil {
-			return nil, err
+			if strings.HasPrefix(stderr, "RTNETLINK answers: File exists") {
```
[the remark this answers is not in the document]
Comment (dcbw) [reply go-controller/pkg/cluster/management-port_linux.go]: yeah this is terrible :) netlink FTW (ouch have I ever said that before? bad me...)

[the remark this answers is not in the document]
Comment (dcbw) [reply go-controller/pkg/cluster/management-port_linux.go]: also, if we do netlink we can RouteReplace() which uses NLM_F_REPLACE which should work for add & replace both.

[the remark this answers is not in the document]
Comment (russellb) [reply go-controller/pkg/cluster/management-port_linux.go]: to clarify ... are you willing to take a hack like this, or would you like to block on a port to netlink?

File: go-controller/pkg/cluster/management-port_linux.go (hunk 1 above)
Comment (russellb) [reply go-controller/pkg/cluster/management-port_linux.go]: Maybe this was added as precautionary after adding the one above?  not sure ... we can try with just the first one and see if it still works?

File: go-controller/pkg/cluster/management-port_linux.go [hunk 3]
```diff
@@ -55,9 +56,14 @@ func CreateManagementPort(nodeName string, localSubnet *net.IPNet, clusterSubnet
...
+				logrus.Warningf("Ignoring error %s from 'route add %s via %s' - already added via IPv6 RA?",
+					strings.TrimSpace(stderr), subnet, routerIP)
```
[the remark this answers is not in the document]
Comment (russellb) [reply go-controller/pkg/cluster/management-port_linux.go]: or even Debugf?  it seems pretty harmless to ignore this ...

File: go-controller/pkg/cluster/management-port_linux.go (hunk 1 above)
Comment (russellb) [reply go-controller/pkg/cluster/management-port_linux.go]: or how about leaving it and just making it a debug message?  Ignoring this seems pretty harmless ...

**Comment by russellb** [comment]
This is now included in a larger IPv6 PR - https://github.com/ovn-org/ovn-kubernetes/pull/968

**Pull request closed without merging**